devito/finite_differences/differentiable.py

```diff
@@ -145,24 +187,47 @@ def __eq__(self, other):
         return super(Differentiable, self).__eq__(other) and\
             all(getattr(self, i, None) == getattr(other, i, None) for i in self._state)
 
+    def index(self, dim):
+        for i, d in zip(self.indices, self.dimensions):
+            if d is dim:
+                return d
```

Comment:
Im totally confused after our discussion on slack. SHould this return `i` or `d` ?